spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/RecordFilterStrategy.java

```diff
@@ -49,9 +51,10 @@ public interface RecordFilterStrategy<K, V> {
 	 * @return the filtered records.
 	 * @since 2.8
 	 */
+	@SuppressWarnings("unchecked")
 	default List<ConsumerRecord<K, V>> filterBatch(List<ConsumerRecord<K, V>> records) {
-		records.removeIf(this::filter);
-		return records;
+		var recordsArray = records.stream().filter(record -> !this.filter(record)).toArray(ConsumerRecord[]::new);
+		return Arrays.asList(recordsArray);
```

Comment:
Why do we need this change?
Why that `removeIf()` doesn't work?

Reply:
I had `removeIf` here before, but tests failed because it was throwing an exception 

```
Caused by: java.lang.UnsupportedOperationException: remove
	at java.base/java.util.Iterator.remove(Iterator.java:102) ~[?:?]
	at java.base/java.util.Collection.removeIf(Collection.java:582) ~[?:?]
	at org.springframework.kafka.listener.adapter.RecordFilterStrategy.filterBatch(RecordFilterStrategy.java:57) ~[main/:?]
```

Reply:
Stepping through it in a debugger showed that `Arrays$ArrayItr` does not support `remove`

![image](https://github.com/user-attachments/assets/5d165849-54f8-4f02-9128-30290fcb1219)

Reply:
Therefore my request for the `new ArrayList()` instead.